internal/utils/endorsement.go

```diff
@@ -3,14 +3,22 @@ package utils
 import (
 	"fmt"
 
-	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	pmsp "github.com/hyperledger/fabric-protos-go/msp"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
 	"github.com/hyperledger/fabric/bccsp/factory"
-	"github.com/hyperledger/fabric/common/cauthdsl"
 	"github.com/hyperledger/fabric/msp"
 	"github.com/hyperledger/fabric/msp/mgmt"
+	"github.com/pkg/errors"
 )
 
-func NewPolicyEvaluator() *PolicyEvaluator {
+type PolicyEvaluatorInterface interface {
```

Comment:
we have an interface here to add mocks so in `registry_test.go` we can better control the behaviour of the policy evaluator. I think this is also the normal way to go in fabric. Once you implement a component that is composed of other components, you better use interfaces so you can concentrate on the implementation and testing of the component you are developing.

ok, the main reason for the change here is that the previous implementation of `EvaluateIdentity` relied on a identityDeserializer provided by default msp provider (not sure if this is the right name?!?! anyway). In particular, the old version called the following code to get the deserializer.

```go
manager := mgmt.NewDeserializersManager(factory.GetDefault())
eturn &PolicyEvaluator{
		IdentityDeserializer: manager.GetLocalDeserializer(),
``` 

However, using this code a chaincode environment we get a deserializer that cannot parse the identities simply because the chaincode environment has no access to the msp information that a peer has. 

Therefore, this change here. What we do now is essentially, unmarshal the policy and the serialized identity and check if the msp id of is part of the policy.

Reply:
will add a few comments

Reply:
Not sure what you mean with different implementation of this evaluator?

Reply:
Actually, i think i know where the disconnect comes. Originally before lightweight the chaincode endorsement policy was really policy which decided on validation rules for attestations, i.e., what in our proposal we call `enclave endorsement` policies.  However, now we have this `enclave endorsement policy` separate from the `validation endorsement policy` and the endorsement policy of the chaincode is the latter whereas the former is for MVP `designated enclave` policy implicitly defined  (and for post-mvp specified in some yet to be determined fashion).  What you reference/verify here is hence against the `validation endorsement policy`.  However, what we do have to verify in errc is the `enclave endorsement policy`.  Hence my reference to above that i was speculating that we might have (eventually) multiple implementations depending on what enclave endorsement policy we support. Not that i think this is really necessary and i now also understand why you have an interface even if there is only one implementation, the current implemention imho is doing the wrong stuff (at least as used by ercc)